LogView improvements

takase1121 opened the pull request:
tl;dr - make LogView more fancy.

https://user-images.githubusercontent.com/20792268/130354265-87e1287b-cd60-4ee9-a52f-ec65a3984fcc.mp4

- hide filename and stack traces by default
- copy individual log entry
- open log as a file

Some discussions:
- some people probably want to expand everything by default
- Manipulating logs with keyboard
- text wrapping (or maybe a horizontal scrollbar)

franko commented:
The idea is nice but I have some text layout problems:

![image](https://user-images.githubusercontent.com/433545/130498402-cb8690e0-e0d1-4786-9a0c-a756a5215d1a.png)

Note the ellipsis appearing on the right "...ancesco", that looks wrong. I am wondering if it isn't better to leave the ellipses out of the implementation.

franko commented:
My bad, I see the ellipsis like in "...ancesco" are already present in the standard LogView. On the other side the problem stands out more in the new implementation because it nicely shows the hidden text using the '+' button but the text with the ellipsis looks sort of out-of-place, especially since we have room of the right to display more character. Also as the hidden line is shown below in its own line it should be truncated with the ellipsis on the right, not on the left.

takase1121 commented:
Yes, there are a few bugs that I missed. I squashed them.

> Also as the hidden line is shown below in its own line it should be truncated with the ellipsis on the right, not on the left.

I removed elipsis completely (since we have a pending horizontal scroll PR, we could use that), if you still see elipsis, that is not my doing.

franko commented:
Great, now it all looks perfect to me, thank you @takase1121.
